paws.common/R/config.R

```diff
@@ -177,19 +177,54 @@ get_profile_name <- function(profile = "") {
 
 # Gets the instance metadata by making an http request.
 get_instance_metadata <- function(query_path = "") {
-
+  token_ttl <- '21600' # same approach as in boto3: https://github.com/boto/botocore/blob/master/botocore/utils.py#L376
   # Do not get metadata when the disabled setting is on.
   if (trimws(tolower(get_env("AWS_EC2_METADATA_DISABLED"))) %in% c("true", "1")) {
     return(NULL)
   }
-
+  # Get token timeout for IMDSv2 tokens
+  token <-  "" # Token to be used in case of more secure IMDSv2 authentication
+  #try IMDSv2  (more information): https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html
+   metadata_token_url <-  file.path(
+    "http://169.254.169.254/latest/api/token"
+  )
+ metadata_token_request <-
+    new_http_request(
+        "PUT", 
+        metadata_token_url, 
+        timeout = 1, 
+        header=c("X-aws-ec2-metadata-token-ttl-seconds"= token_ttl)
+    )
+
+  metadata_token_response <- tryCatch(
+    {
+      issue(metadata_token_request)
+    },
+    error = function(e) {
+      NULL
+    }
+  )
+  if (!(is.null(metadata_token_response)) && metadata_token_response$status_code == 200) {
+      if (length(metadata_token_response["body"])>0) {
+          token=rawToChar(metadata_token_response["body"])
+      } 
+  }
   metadata_url <- file.path(
     "http://169.254.169.254/latest/meta-data",
     query_path
   )
+  if (token!="") {
+  metadata_request <-
+    new_http_request(
+        "GET", 
+        metadata_url, 
+        timeout = 1, 
+        header=c("X-aws-ec2-metadata-token"= token)
+    )
+  } else { # use IMDSv1 in case IMDSv2 is not available - not recommended - very insecure
```

Comment:
is there a standard way in the AWS SDKs to force use of IMDSv2 (or v1), rather than defaulting to v1? this comment suggests that defaulting to v1 is a bad practice. if not, no change needed.

nit: I would make a separate get_metadata_request_imds_v1 and get_metadata_request_imds_v2 to make it easier to follow this logic.

Reply:
As far as I know one can only configure it on the resource itself (EC2, Sagemaker Notebook etc.) to use IMDSv2-only, so you cannot just do it from the API itself.

Hence, I think splitting it up into get_metadata_request_imds_v1 and get_metadata_request_imds_v2 would not be really possible. I put them into one, because IMDSv2 is in essence just an additional HTTP request beforehand.

We could have a function fetch_metadata_token to fetch the token and imds_request that adds the token if is available or not (see also: https://github.com/boto/botocore/blob/master/botocore/utils.py#L569). 

Currently, I tried to integrate it into the existing logic in PAWS without rewriting what was already defined before.

Reply:
Ok sounds good but in that case I’d get rid of “not recommended - very insecure” because someone who doesn’t know that background could delete this code inappropriately.

Reply:
ok makes sense what you say. Maybe I will put security recommendations better in the documentation - the chance that somebody reads the documentation are higher ;)

Reply:
Eh, that's not necessary.

Reply:
ok removed the comment and added a reference to the AWS documentation on IMDS